types/staking.go

```diff
@@ -160,3 +160,62 @@ type StakingHooks interface {
 	AfterDelegationModified(ctx Context, delAddr AccAddress, valAddr ValAddress)
 	BeforeValidatorSlashed(ctx Context, valAddr ValAddress, fraction Dec)
 }
+
```

Comment:
Are these hooks generally useful for other staking implementations? Or is this something that could/should live in `x/staking`?

Reply:
hmm yeah good point this belongs in x/staking